### PR TITLE
Update rector to 0.12.23

### PR DESCRIPTION
**Pull request by kenjis**
Supersedes #57

- update rector
- update github workflow files

### CODE REVIEW

File: rector.php [hunk 1]
```diff
@@ -97,29 +90,30 @@
...
+    $rectorConfig->rule(UnnecessaryTernaryExpressionRector::class);
+    $rectorConfig->rule(AddPregQuoteDelimiterRector::class);
+    $rectorConfig->rule(SimplifyRegexPatternRector::class);
+    $rectorConfig->rule(FuncGetArgsToVariadicParamRector::class);
+    $rectorConfig->rule(MakeInheritedMethodVisibilitySameAsParentRector::class);
+    $rectorConfig->rule(SimplifyEmptyArrayCheckRector::class);
+    $rectorConfig->rule(NormalizeNamespaceByPSR4ComposerAutoloadRector::class);
+    $rectorConfig
+        ->ruleWithConfiguration(TypedPropertyRector::class, [
+            TypedPropertyRector::INLINE_PUBLIC => false,
```
Comment (MGatner) [inline rector.php:117]: 👀

**Review by MGatner — APPROVED** [review]

**Comment by MGatner** [comment]
PHPStan seems to be choking on the `setting()` return in `Task::lastRun()` causing it to expect `'--'` returns every time. I'm not sure how to handle this other than ignoring the complicit line in **TaskTest.php**.

File: rector.php (hunk 1 above)
Comment (kenjis) [reply rector.php:117]: Without this, BC was made.

**Comment by kenjis** [comment]
PHPStan never knows `setting()` changes the state.
It seems there is no way other than ignoring it.

**Pull request merged**